vectors.go

```diff
@@ -0,0 +1,284 @@
+package pinecone
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+)
+
+// DescribeIndexStatsParams represents the parameters for a describe index stats request.
+type DescribeIndexStatsParams struct {
+	Filter map[string]any `json:"filter"`
+}
+
+// DescribeIndexStatsResponse represents the response from a describe index stats request.
+type DescribeIndexStatsResponse struct {
+	Namespaces       map[string]*VectorCount `json:"namespaces"`
+	Dimensions       int                     `json:"dimensions"`
+	IndexFullness    float32                 `json:"index_fullness"`
+	TotalVectorCount int                     `json:"totalVectorCount"`
```

Comment:
```suggestion
	Namespaces       map[string]*VectorCount `json:"namespaces"`
	Dimensions       int64                     `json:"dimension"`
	IndexFullness    float32                 `json:"indexFullness"`
	TotalVectorCount int64                     `json:"totalVectorCount"`
```

![image](https://user-images.githubusercontent.com/11081491/236595470-f8bb25a6-8099-44c1-ab31-72ce08ad485a.png)

Reply:
done